test/parallel_testsuite.py

```diff
@@ -17,6 +17,9 @@
 
 
 def run_test(test):
+  # Prevent the "Terminate batch job (Y/N)?" dialog from intercepting Ctrl+C
+  # on Windows.
+  os.close(0)
```

Comment:
We could also do this before we create the worker pool, which might be less racey?

Reply:
I tried that in `init_worker`, and it also interrupts, but it makes the output a lot worse. I suppose multiprocessing itself sets up the stdio for each run rather than at the beginning?

With current change:

```
test_safe_stack_pthread (test_core.core0) ... ok (2.82s)
test_pthread_unhandledrejection (test_core.core0) ... ok (2.80s)
.. exit
test_pthread_thread_local_storage (test_core.core0) ... ok (3.10s)
Process SpawnPoolWorker-8:
Process SpawnPoolWorker-4:
Process SpawnPoolWorker-7:
Process SpawnPoolWorker-1:
Process SpawnPoolWorker-2:
test_pthread_stdout_after_main (test_core.core0) ... ERROR
test_sqlite_pthreads (test_core.core0) ... ERROR
runner:WARNING: KeyboardInterrupt
Terminate batch job (Y/N)? 
```

With doing it in `init_worker`:

```
test_pthread_unhandledrejection (test_core.core0) ... ok (3.47s)
test_pthread_thread_local_storage (test_core.core0) ... ok (3.61s)
test_pthread_proxying_refcount (test_core.core0) ... ok (3.69s)
Traceback (most recent call last):Traceback (most recent call last):
test_pthread_stdout_after_main (test_core.core0) ... ERROR
  File "C:\Users\me\Documents\emscripten\\emcc.py", line 23, in <module>test_sqlite_pthreads (test_core.core0) ... ERROR
  File "C:\Users\me\Documents\emscripten\\emcc.py", line 41, in <module>
    from tools.toolchain_profiler import ToolchainProfiler    import emscripten
  File "C:\Users\me\Documents\emscripten\tools\toolchain_profiler.py", line 7, in <module>  File "C:\Users\me\Documents\emscripten\emscripten.py", line 22, in <module>
    import logging    from tools import building
  File "C:\Python38\lib\logging\__init__.py", line 26, in <module>  File "C:\Users\me\Documents\emscripten\tools\building.py", line 22, in <module>
Terminate batch job (Y/N)?
Terminate batch job (Y/N)?
runner:WARNING: KeyboardInterrupt
    import sys, os, time, io, re, traceback, warnings, weakref, collections.abc    from . import shared
  File "<frozen importlib._bootstrap>", line 991, in _find_and_load  File "C:\Users\me\Documents\emscripten\tools\shared.py", line 28, in <module>
Terminate batch job (Y/N)?

    from . import colored_logger
  File "<frozen importlib._bootstrap>", line 975, in _find_and_load_unlocked
  File "<frozen importlib._bootstrap>", line 991, in _find_and_load
  File "<frozen importlib._bootstrap>", line 671, in _load_unlocked
  File "<frozen importlib._bootstrap>", line 975, in _find_and_load_unlocked
  File "<frozen importlib._bootstrap_external>", line 779, in exec_module
  File "<frozen importlib._bootstrap>", line 671, in _load_unlocked
  File "<frozen importlib._bootstrap_external>", line 874, in get_code
  File "<frozen importlib._bootstrap_external>", line 779, in exec_module
  File "<frozen importlib._bootstrap_external>", line 972, in get_data
  File "<frozen importlib._bootstrap_external>", line 874, in get_code
KeyboardInterrupt
  File "<frozen importlib._bootstrap_external>", line 972, in get_data
KeyboardInterrupt
^CTerminate batch job (Y/N)?
^CTerminate batch job (Y/N)?
```

Reply:
Oh wait, you said before creating the entire pool, not before each worker. Let me try that.

Reply:
Yeah same effect as with doing it in `init_worker` - it works, but the output is similarly terrible.

Reply:
Ahh I see the test failures precisely due to racy `close()`...

Ugh. I don't want that verbose output, but I just realised that the only reason it's shorter with current approach is because of the `os.close()` throwing and erroring out those tests early 😅

Reply:
I'll look into something better.

Reply:
Fine, I'll just do the init_worker for now... Better than nothing.

Reply:
I'm not sure I really understand the different issues at play but if you say this is better then LGTM!